feat: 支持模型供应商头像裁剪功能

vectorstone opened the pull request:
<!-- Template from https://github.com/kubevirt/kubevirt/blob/main/.github/PULL_REQUEST_TEMPLATE.md?-->
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/CherryHQ/cherry-studio/blob/main/CONTRIBUTING.md
-->

### What this PR does

Before this PR:

After this PR:

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->

Fixes #

### Why we need it and why it was done in this way

The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Breaking changes

<!-- optional -->

If this PR introduces breaking changes, please describe the changes and the impact on users.

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Documentation: A [user-guide update](https://docs.cherry-ai.com) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature.

### Release note

<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->

```release-note
1. **User Experience Enhancement**:
   - Introduced image editor allowing users to crop and adjust images
   - Preserves GIF files as-is to avoid breaking animation effects

2. **Code Simplification**:
   - Removed dependency on complex `compressImage` function
   - Simplified Blob/File type handling logic
   - Reduced code complexity

3. **Separation of Concerns**:
   - Image compression/editing functionality moved to dedicated `ProviderAvatarEditor` component
   - Clearer responsibility separation
```

vectorstone committed:
支持模型供应商头像裁剪功能 https://github.com/CherryHQ/cherry-studio/issues/9679

EurFelux commented on src/renderer/src/components/ImageEditor/ImageEditor.tsx:
```diff
@@ -0,0 +1,299 @@
+  imageSrc,
+  onCancel,
+  onConfirm,
+  title = '编辑图片',
```
i18n

EurFelux commented on src/renderer/src/components/ImageEditor/ImageEditor.tsx:
```diff
@@ -0,0 +1,299 @@
+import { Button, message, Modal, Slider, Space, Upload } from 'antd'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import Cropper, { ReactCropperElement } from 'react-cropper'
+// import { useTranslation } from 'react-i18next' // 暂时不使用翻译
```
要用

vectorstone replied:
已修正

EurFelux commented on src/renderer/src/components/ImageEditor/ImageEditor.tsx:
```diff
@@ -0,0 +1,299 @@
+
+  const handleConfirm = useCallback(async () => {
+    if (!cropperRef.current?.cropper) {
+      message.error('图片编辑器未准备就绪')
```
```suggestion
      window.message.error('i18n here')
```

vectorstone replied:
已修正

EurFelux commented on tall:
```diff
@@ -0,0 +1,10 @@
+[33m30bd87e4[m[33m ([m[1;36mHEAD[m[33m -> [m[1;32m250830-gavin-dialog-html[m[33m, [m[1;31morigin/250830-gavin-dialog-html[m[33m)[m 修复issue https://github.com/CherryHQ/cherry-studio/issues/9635,支持<details/>和<summary/>标签渲染展示,后期可以考虑做成配置化,允许用户自定义标签展示逻辑,或者扩展rehypeRaw插件标签识别逻辑,支持任意标签的渲染展示
```
这文件是什么？

vectorstone committed:
删除无用组件ImageEditor以及无用文件tall,修改代码中硬编码文案描述,改用i18n

vectorstone committed:
修改使用指南描述

vectorstone committed:
删除无用i18n文案描述

vectorstone committed:
删除无引用的i18n文案描述

vectorstone commented on src/renderer/src/components/ImageEditor/ImageEditor.tsx:
```diff
@@ -0,0 +1,299 @@
+                <SectionTitle>旋转和翻转</SectionTitle>
+                <Space size="middle">
+                  <Button icon={<RotateLeftOutlined />} onClick={handleRotateLeft} size="small">
+                    左转
```
感谢提醒,已使用i18n替换代码中硬编码的文案描述

EurFelux commented:
有个预览裁剪后头像样式的地方会更好

Commit:
Arch Linux上yar dev启动项目的时候会出现system-ocr包错误的提示

Commit:
Merge remote-tracking branch 'origin' into 250831-gavin-model-provider-avatar-editor

vectorstone committed:
Merge remote-tracking branch 'origin/main' into 250831-gavin-model-provider-avatar-editor

# Conflicts:
#	package.json
#	src/renderer/src/pages/settings/ProviderSettings/AddProviderPopup.tsx

vectorstone committed:
Merge branch '250831-gavin-model-provider-avatar-editor' of github.com:vectorstone/cherry-studio into 250831-gavin-model-provider-avatar-editor

# Conflicts:
#	src/renderer/src/pages/settings/ProviderSettings/AddProviderPopup.tsx

Commit:
裁剪头像增加预览功能,修改布局样式

Commit:
恢复误删除的文件

Commit:
恢复误删除的文件

vectorstone commented:
> 有个预览裁剪后头像样式的地方会更好

1.增加实时头像预览功能
2.调整样式布局避免出现填充不足或者不居中的问题
<img width="643" height="759" alt="image" src="https://github.com/user-attachments/assets/7e537683-81bf-4f6a-b3e6-c20e1ff02292" />

Commit:
替换ant组件以及styled-components

Commit:
解决导入以及类型的问题

Commit:
Merge remote-tracking branch 'origin/main' into 250831-gavin-model-provider-avatar-editor

Commit:
react-cropper挪动到开发依赖中

Commit:
react-cropper挪动到开发依赖中

EurFelux committed:
Merge remote-tracking branch 'upstream/main' into 250831-gavin-model-provider-avatar-editor

EurFelux committed:
fix: replace window.message with window.toast for error notifications

EurFelux committed:
refactor(image-editor): replace custom cropper styles with cropper.css

The custom cropper styles in ProviderAvatarEditor.module.css were removed and replaced with the standard cropper.css file. This change simplifies maintenance by using the official Cropper.js styles instead of maintaining custom overrides.

EurFelux commented:
I redesigned the modal.

<img width="1794" height="1406" alt="image" src="https://github.com/user-attachments/assets/26a1de75-0c3f-4c3d-88d9-d9c30b3aac34" />

EurFelux committed:
fix(i18n): remove trailing colon from usage guide text

EurFelux committed:
refactor(avatar-editor): migrate to new modal component and improve UI

- Replace custom modal implementation with @heroui/react Modal component
- Restructure layout using grid and flexbox for better responsiveness
- Increase max cropped image dimensions from 60x60 to 200x200
- Update button styles and interactions to match design system
- Move list-style CSS rule to nested scope for consistency

EurFelux committed:
feat(theme): add foreground color calculation for primary theme

Calculate and set foreground color based on primary theme color to ensure proper contrast

EurFelux committed:
style(css): fix indentation and formatting in css files

EurFelux committed:
docs(i18n): add missing german translations for avatar editor

beyondkmp commented on package.json:
```diff
@@ -83,10 +83,13 @@
     "@strongtz/win32-arm64-msvc": "^0.4.7",
+    "cropperjs": "^2.0.1",
     "express": "^5.1.0",
+    "faiss-node": "^0.5.1",
```
we have deleted this package.

beyondkmp commented:
> [!NOTE]
> This issue/comment/review was translated by Claude.

There are issues with the package.json changes in this PR, many packages are unused and were added?

---
<details>
<summary>Original Content</summary>

这个PR的package.json改动有问题，很多包都没有用，还添加了？

</details>

EurFelux commented:
> There are issues with the package.json changes in this PR, many packages are unused and were added?

We can remove packages that don't affect the functionality. Actually, I don't understand why he added these packages except `react-cropper`.

beyondkmp commented:
> [!NOTE]
> This issue/comment/review was translated by Claude.

Should there be a problem with merging the main branch? Or the main branch has removed these libraries, but this PR didn't remove them, causing this issue.

---
<details>
<summary>Original Content</summary>

应该是merge主分支有问题？或者主分支有删除了这些库了，这个PR没有删除导致的。

</details>

EurFelux committed:
chore: remove unused dependencies from package.json and yarn.lock

EurFelux committed:
style(ProviderAvatarEditor): remove flex-wrap from button container

EurFelux committed:
chore: move cropperjs from dependencies to devDependencies

EurFelux committed:
style(ImageEditor): adjust button container alignment in avatar editor

EurFelux reviewed (approved):
I have tested this feature in both the macOS development and production environments, and no issues were found.

GeorgeDong32 commented on .gitignore:
```diff
@@ -57,6 +57,8 @@ local
 .codebuddy/*
 .zed/*
 CLAUDE.local.md
+tall
```
这两行无关的是不是去掉比较好？

EurFelux replied:
true

GeorgeDong32 reviewed (approved):
在Windows上测试没问题

EurFelux committed:
chore: remove unused entries from gitignore
